Extension/test/integrationTests/MockDebugger/debugAdapterDescriptorFactory.ts

```diff
@@ -0,0 +1,47 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from "vscode";
+import * as path from 'path';
+
+// import * as util from '../src/common' <- DO NOT USE. Also do not use anything with relative paths, it will break during hijacking in test/debug.integration.test.ts
+
+abstract class DebugAdapterDescriptorFactoryWithContext implements vscode.DebugAdapterDescriptorFactory {
```

Comment:
Why not abstract this to another file so its not duplicated?

Reply:
The .js file is copied during the pipeline, trying to remove any relative imports as possible.